moveit_ros/hybrid_planning/constraint_solver_plugins/src/handle_imminent_collision.cpp

```diff
@@ -70,16 +70,18 @@ HandleImminentCollision::solve(robot_trajectory::RobotTrajectory local_trajector
                                std::vector<moveit_msgs::msg::Constraints> local_constraints,
                                std::shared_ptr<moveit_msgs::action::LocalPlanner::Feedback> feedback)
 {
-  // Clone current planning scene
+  // Read current planning scene
   planning_scene_monitor_->updateFrameTransforms();
   planning_scene_monitor_->lockSceneRead();  // LOCK planning scene
-  planning_scene::PlanningScenePtr planning_scene = planning_scene::PlanningScene::clone(
-      planning_scene_monitor_->getPlanningScene());  // TODO remove expensive planning scene cloning
-  planning_scene_monitor_->unlockSceneRead();        // UNLOCK planning scene
+  planning_scene::PlanningScenePtr planning_scene = planning_scene_monitor_->getPlanningScene();
```

Comment:
This won't do what you want (the planning scene instance may change after unlocking so your collision check won't be protected). The recommended way to do this would be to initialize a `LockedPlanningSceneRO` which gives you read-locked access to the planning scene until it goes out of scope. http://docs.ros.org/en/melodic/api/moveit_ros_planning/html/classplanning__scene__monitor_1_1LockedPlanningSceneRO.html